sql: remove experimental_partial_indexes setting and disallow inverted partial indexes

mgartner opened the pull request:
#### sql: disallow inverted partial indexes

Inverted partial indexes are not fully supported. This commit disallows
them from being created.

Fixes #52025

Release note: None

#### sql: removes experimental_partial_indexes setting

This commit removes the `experimental_partial_indexes` session and
cluster setting, now that partial indexes are supported.

Fixes #50217

Release note: None

mgartner committed:
sql: disallow inverted partial indexes

Inverted partial indexes are not fully supported. This commit disallows
them from being created.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53330)
<!-- Reviewable:end -->

mgartner committed:
sql: removes experimental_partial_indexes setting

This commit removes the `experimental_partial_indexes` session and
cluster setting, now that partial indexes are supported.

Release note: None

mgartner commented:
TFTR!

bors r+
